core/src/main/java/org/apache/carbondata/core/datastore/chunk/DimensionColumnPage.java

```diff
@@ -97,21 +96,19 @@
    * @param compareValue value to compare
    * @return compare result
    */
-  int compareTo(int rowId, byte[] compareValue);
+  int compareTo(int rowId, Object compareValue);
 
   /**
    * below method will be used to free the allocated memory
    */
   void freeMemory();
 
-  /**
-   * to check whether the page is adaptive encoded
-   */
   boolean isAdaptiveEncoded();
 
   /**
-   * to get the null bit sets in case of adaptive encoded page
+   * method return presence meta which represents
+   * rowid for which values are null or not null
+   * @return
    */
-  BitSet getNullBits();
-
+  PresenceMeta getPresentMeta();
```

Comment:
This method purpose is not clear, seems to be PresenceMeta is not very informative as per the comment which you provided on the method